### PR TITLE
fix e2e test following removal of restricted admin role

### LINKED ISSUE #12689: Fix e2e tests following the removal of the Restricted Admin role
**Setup**
- Rancher version: `2.11-head`
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

- Our e2e tests are failing on the `auth` part: https://github.com/rancher/dashboard/actions/runs/12036272266/job/33593793662?pr=12681

Seems the culprit is the removal of the `Restricted Admin` role
Screenshot 2024-11-27 at 11 56 09
Screenshot 2024-11-27 at 11 56 04

**To Reproduce**

**Result**
- pipeline is failing

**Expected Result**

- pipeline should succeed

**Screenshots**

**Additional context**

- seems that the backend has done the removal of the `Restricted Admin` role

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #12689 

### Occurred changes and/or fixed issues

- fix e2e test following removal of restricted admin role

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Review by cnotv — APPROVED** [review]
LGTM

File: cypress/e2e/tests/pages/users-and-auth/users.spec.ts [hunk 1]
```diff
@@ -122,9 +107,8 @@ describe('Users', { tags: ['@usersAndAuths', '@adminUser'] }, () => {
     const mgmtUserEditPo = new MgmtUserEditPo();
 
     mgmtUserEditPo.globalRoleBindings().globalOptions().then((list) => {
-      expect(list.length).to.eq(4);
+      expect(list.length).to.eq(3);
       expect(list[0]).to.eq('Administrator');
-      expect(list[1]).to.eq('Restricted Administrator');
```
Comment (cnotv) [inline cypress/e2e/tests/pages/users-and-auth/users.spec.ts:127]: After this change, all the other indexes must be updated accordingly, or?

Cypress dashboard: `fe43b3ff-a329-4c38-8d8e-cb81336c319d`

Screenshot 2024-11-27 at 15 53 03

Comment (aalves08) [reply cypress/e2e/tests/pages/users-and-auth/users.spec.ts:127]: updated the PR

**Comment by cnotv** [comment]
Still failing one test, but I do not have unfortunately anything to point :( 

Screenshot 2024-11-27 at 16 57 17

**Comment by aalves08** [comment]
> Still failing one test, but I do not have unfortunately anything to point :(
> 
> Screenshot 2024-11-27 at 16 57 17

flakiness.... retrying the job

**Review by cnotv — APPROVED** [review]
pre-CI LGTM 😄

**Pull request merged**